src/api/app/controllers/person/token_controller.rb

```diff
@@ -0,0 +1,42 @@
+require 'xmlhash'
+module Person
+  class TokenController < ApplicationController
+    before_action :set_user, only: [:index, :create, :delete]
+
+    # GET /person/<login>/token
+    def index
+      authorize @user, :show?
+
+      @list = @user.service_tokens
+    end
+
+    # POST /person/<login>/token
+    def create
+      authorize @user, :update?
+
+      pkg = nil
+      if params[:project] || params[:package]
+        pkg = Package.get_by_project_and_name(params[:project], params[:package])
+      end
+      @token = Token::Service.create(user: @user, package: pkg)
+    end
+
+    # DELETE /person/<login>/token/<id>
+    def delete
+      authorize @user, :update?
+
+      token = Token::Service.where(user_id: @user.id, id: params[:id]).first
```

Comment:
``find_by(...)``?

Reply:
`@user.service_tokens.find!(params[:id])`

then you can skip `render_error` below right?